server/balancer.go

```diff
@@ -105,15 +105,21 @@ func (cb *capacityBalancer) SelectToStore(stores []*StoreInfo, excluded map[uint
 	return resultStore
 }
 
-func (cb *capacityBalancer) SelectBalanceRegion(stores []*StoreInfo, regions *RegionsInfo) (*metapb.Region, *metapb.Peer) {
+func (cb *capacityBalancer) SelectBalanceRegion(stores []*StoreInfo, cluster *ClusterInfo) (*metapb.Region, *metapb.Peer) {
 	store := cb.SelectFromStore(stores, true)
 	if store == nil {
 		return nil, nil
 	}
 
 	// Random select one leader region from store.
 	storeID := store.store.GetId()
-	region := regions.randRegion(storeID)
+	region := cluster.regions.randRegion(storeID)
+
+	// If region peer count is less than max peer count, no need to do balance.
```

Comment:
I think we should pass a filter in randRegion to filter these regions when selecting.